src/backend/pipeline/cont_analyze.c

```diff
@@ -1560,7 +1560,7 @@ transformContSelectTargetList(ParseState *pstate, List *tlist)
 				 * CV's aggregate types will not have been changed. This can cause a
 				 * signature resolution error when resolving the transout function.
 				 */
-				if (IsContQueryProcess())
+				if (IsContQueryProcess() || IsContQueryAdhocProcess())
```

Comment:
Maybe we should just change the `IsContQueryProcess()` macro to return true for ad-hoc procs as well?